sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp

```diff
@@ -0,0 +1,116 @@
+//==--- user_defined_reductions.hpp -- SYCL ext header file -=--*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/detail/defines.hpp>
+#include <sycl/group_algorithm.hpp>
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext::oneapi::experimental {
+
+// ---- reduce_over_group
+template <typename GroupHelper, typename T, typename BinaryOperation>
+sycl::detail::enable_if_t<(is_group_helper_v<GroupHelper>), T>
```

Comment:
> (@aelovikov-intel) Why not make `std::decay_t` part of `is_group_helper`?

Applied in https://github.com/intel/llvm/pull/7587/commits/446f6722292aa0bf8f7061e64a8e93eb5ff3410e